Library/Homebrew/test/requirement_spec.rb

```diff
@@ -0,0 +1,237 @@
+      let(:tags) { ["bar"] }
+
+      its(:tags) { are_expected.to eq(tags) }
+    end
+
+    context "multiple tags" do
+      let(:tags) { ["bar", "baz"] }
+
+      its(:tags) { are_expected.to eq(tags) }
+    end
+
+    context "symbol tags" do
+      let(:tags) { [:build] }
+
+      its(:tags) { are_expected.to eq(tags) }
+    end
+
+    context "symbol and string tags" do
+      let(:tags) { [:build, "bar"] }
+
+      its(:tags) { are_expected.to eq(tags) }
+    end
+  end
+
+  describe "#fatal?" do
+    subject { Class.new(described_class) { fatal true }.new }
+
+    it { is_expected.to be_fatal }
+  end
+
+  describe "#satisfied?" do
+    context "#satisfy with block and build_env returns true" do
+      let(:klass) do
+        Class.new(described_class) do
+          satisfy(build_env: false) do
+            true
+          end
+        end
+      end
+
+      it { is_expected.to be_satisfied }
+    end
+
+    context "#satisfy with block and build_env returns false" do
+      let(:klass) do
+        Class.new(described_class) do
+          satisfy(build_env: false) do
+            false
+          end
+        end
+      end
+
+      it { is_expected.not_to be_satisfied }
+    end
+
+    context "#satisfy returns true" do
+      let(:klass) do
+        Class.new(described_class) do
+          satisfy true
+        end
+      end
+
+      it { is_expected.to be_satisfied }
+    end
+
+    context "#satisfy returns false" do
+      let(:klass) do
+        Class.new(described_class) do
+          satisfy false
+        end
+      end
+
+      it { is_expected.not_to be_satisfied }
+    end
+
+    context "#satisfy with block returns true" do
+      let(:klass) do
+        Class.new(described_class) do
+          satisfy { true }
+        end
+      end
+
+      it "sets up build environment by default" do
+        expect(ENV).to receive(:with_build_environment) do
+          yield true if block_given?
+          true
+        end
+
+        expect(subject).to be_satisfied
+      end
+    end
+
+    context "#satisfy with block and build_env returns true" do
+      let(:klass) do
+        Class.new(described_class) do
+          satisfy(build_env: false) do
+            true
+          end
+        end
+      end
+
+      it "build environment can be disabled" do
+        expect(ENV).not_to receive(:with_build_environment)
+
+        expect(subject).to be_satisfied
+      end
+    end
+  end
+
+  context "#satisfy with block and build_env returns path" do
+    let(:klass) do
+      Class.new(described_class) do
+        satisfy { path }
+      end
+    end
+
+    let(:path) { Pathname.new("/foo/bar/baz") }
+
+    it "infers path from #satisfy result" do
+      expect(ENV).to receive(:with_build_environment) do
+        yield path if block_given?
+        path
+      end
+      expect(ENV).to receive(:append_path).with("PATH", path.parent)
+
+      expect(subject).to be_satisfied
+      subject.modify_build_environment
```

Comment:
`modify_build_environment` depends on `satisfied_result_parent` returning a path, which is set in `satisfied?`, so it has to be called in the same test.